tests/paramgrill.c

```diff
@@ -173,26 +344,406 @@ BMK_benchParam(BMK_result_t* resultPtr,
+*  From Paramgrill End
+*********************************************************/
+
+/* Replicate function of benchMemAdvanced, but with pre-split src / dst buffers, with relevant info to invert it (compressedSizes) passed out. */
+/*BMK_benchMemAdvanced(srcBuffer,srcSize, dstBuffer, dstSize, fileSizes, nbFiles, 0, &cParams, dictBuffer, dictSize, ctx, dctx, 0, "File", &adv); */
+/* nbSeconds used in same way as in BMK_advancedParams_t, as nbIters when in iterMode */
+
+/* if in decodeOnly, then srcPtr's will be compressed blocks, and uncompressedBlocks will be written to dstPtrs? */
+/* dictionary nullable, nothing else though. */
+static BMK_return_t BMK_benchMemInvertible(const void * const * const srcPtrs, size_t const * const srcSizes,
+                        void** dstPtrs, size_t* dstCapacityToSizes, U32 const nbBlocks, 
+                        const int cLevel, const ZSTD_compressionParameters* comprParams,
+                        const void* dictBuffer, const size_t dictBufferSize,
+                        ZSTD_CCtx* ctx, ZSTD_DCtx* dctx,
+                        const BMK_mode_t mode, const BMK_loopMode_t loopMode, const unsigned nbSeconds) {
+    U32 i;
+    BMK_return_t results = { { 0, 0., 0., 0 }, 0 } ;
+    size_t srcSize = 0;
+    void** const resPtrs = malloc(sizeof(void*) * nbBlocks); /* only really needed in both mode. */
+    size_t* const resSizes = malloc(sizeof(size_t) * nbBlocks);
+    int freeDST = 0;
+
+    BMK_advancedParams_t adv = BMK_initAdvancedParams();
+    adv.mode = mode;
+    adv.loopMode = loopMode;
+    adv.nbSeconds = nbSeconds;
+
+    /* resSizes == srcSizes, but modifiable */
+    memcpy(resSizes, srcSizes, sizeof(size_t) * nbBlocks);
+
+    for(i = 0; i < nbBlocks; i++) {
+        srcSize += srcSizes[i];
+    }
+
+    if(!ctx || !dctx || !srcPtrs || ! srcSizes)
+    {
+        results.error = 31;
+        DISPLAY("error: passed in null argument\n");
+        free(resPtrs);
+        free(resSizes);
+        return results;
+    }
+    if(!resPtrs || !resSizes) {
+        results.error = 32;
+        DISPLAY("error: allocation failed\n");
+        free(resPtrs);
+        free(resSizes);
+        return results;
+    }
+
+    /* so resPtr is continuous */ 
+    resPtrs[0] = malloc(srcSize);
+
+    if(!(resPtrs[0])) {
+        results.error = 32;
+        DISPLAY("error: allocation failed\n");
+        free(resPtrs);
+        free(resSizes);
+        return results;
+    }
+
+    for(i = 1; i < nbBlocks; i++) {
+        resPtrs[i] = (void*)(((char*)resPtrs[i-1]) + srcSizes[i-1]);
+    }
+
+    /* allocate own dst if NULL */
+    if(dstPtrs == NULL) {
+        freeDST = 1;
+        dstPtrs = malloc(nbBlocks * sizeof(void*));
+        dstCapacityToSizes = malloc(nbBlocks * sizeof(size_t));
+        if(dstPtrs == NULL) {
+            results.error = 33;
+            DISPLAY("error: allocation failed\n");
+            free(resPtrs);
+            free(resSizes);
+            return results;
```

Comment:
`dstCapacityToSizes` was allocated just before,
and is not freed there, 
so it becomes a memory leak.